pkg/agent/cniserver/sriov.go

```diff
@@ -0,0 +1,144 @@
+// +build linux
+
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cniserver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/types/current"
+	sriovcniutils "github.com/k8snetworkplumbingwg/sriov-cni/pkg/utils"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+
+	// Version v1 of the Kubelet API was introduced in K8s v1.20.
+	// Using version v1alpha1 instead to support older K8s versions.
+	podresourcesv1alpha1 "k8s.io/kubelet/pkg/apis/podresources/v1alpha1"
+
+	"antrea.io/antrea/pkg/agent/util"
+)
+
+const (
+	kubeletPodResourcesPath = "/var/lib/kubelet/pod-resources"
+	kubeletSocket           = "kubelet.sock"
+	connectionTimeout       = 10 * time.Second
+)
+
+type KubeletPodResources struct {
+	resources []*podresourcesv1alpha1.PodResources
+}
+
+func GetPodResourceRequests(podName string, podNamespace string) (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), connectionTimeout)
+	defer cancel()
+
+	conn, err := grpc.DialContext(ctx, path.Join(kubeletPodResourcesPath, kubeletSocket),
+		grpc.WithInsecure(), grpc.WithContextDialer(func(ctx context.Context, addr string) (conn net.Conn, e error) {
+			return util.DialLocalSocket(addr)
+		}))
+	if err != nil {
+		return "", fmt.Errorf("error getting the gRPC client for Pod resources: %v", err)
+	}
+
+	defer conn.Close()
+
+	client := podresourcesv1alpha1.NewPodResourcesListerClient(conn)
+	if client == nil {
+		return "", fmt.Errorf("Error getting the lister client for Pod resources.")
```

Comment:
nit: we don't capitalize error strings (but we do capitalize error logs), so it should be `fmt.Errorf("error getting ...`

a few other occurrences in this file